mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -212,12 +230,139 @@ class LinearRegressionModel private[ml] (
   extends RegressionModel[Vector, LinearRegressionModel]
   with LinearRegressionParams {
 
+  @transient private var trainingSummary: Option[LinearRegressionTrainingSummary] = None
+
+  /**
+   * Gets results summary (e.g. residuals, mse, r-squared ) of model on training set. This method
+   * should only be called on the driver as `summary` is transient.
+   *
+   * An exception is thrown if `trainingSummary == None`.
+   */
+  def summary: LinearRegressionTrainingSummary = trainingSummary match {
+    case Some(summ) => summ
+    case None =>
+      throw new SparkException(
+        "No training summary available for this LinearRegressionModel",
+        new NullPointerException())
+  }
```

Comment:
I have concerns to throw out the exception. Since the whole `LinearRegressionTrainingSummary` is transient, users may mistakenly terminate the program if the models are loaded from files instead of created in training. Thus, we have to ask users to be very carefully in reusing the code in training and online system.